llvm/test/Transforms/PGOProfile/no_global_variables.ll

```diff
@@ -0,0 +1,15 @@
+; RUN: opt < %s -passes=pgo-instr-gen -S | FileCheck %s --check-prefix=GEN-COMDAT
```

Comment:
How does this test behave different than the current status quo?

Reply:
The idea of the test is to check that the assert on line https://github.com/llvm/llvm-project/pull/93421/files#diff-618e0bf5dc7c44fae29843f3be69506a1b0a1409c7f018c61b0acf593f1c4605R408 is never triggered. A module with no global variables (other than the counters inserted during the instrumentation) was prepared to demonstrate that the version variable `__llvm_profile_raw_version` will be generated and it will be inserted in the module just after the corresponding `comdat`. If you think this is a redundant module, I'm going to remove it.

Reply:
if this is just for compile time optimization, then adding these tests isn't necessary

Reply:
ah the commit title is a bit misleading since there is a functional change in that we don't insert __llvm_profile_raw_version if there are no globals. so this test is fine

Reply:
@aeubanks Yes, unfortunately I have no idea how to reflect both changes in the title what is the sign there should be two commits.